src/dlgConnectionProfiles.cpp

```diff
@@ -52,9 +52,85 @@ dlgConnectionProfiles::dlgConnectionProfiles(QWidget * parent)
     profiles_tree_widget->setSelectionMode(QAbstractItemView::SingleSelection);
 
     QAbstractButton* abort = dialog_buttonbox->button(QDialogButtonBox::Cancel);
-    abort->setIcon(QIcon(QStringLiteral(":/icons/dialog-close.png")));
     connect_button = dialog_buttonbox->addButton(tr("Connect"), QDialogButtonBox::AcceptRole);
-    connect_button->setIcon(QIcon(QStringLiteral(":/icons/dialog-ok-apply.png")));
+
+    // Test and set if needed mudlet::mIsIconShownOnDialogButtonBoxes - if there
+    // is already a Qt provided icon on a predefined button, this is probably
+    // the first and best place to test this as the "Cancel" button is a built-
+    // in dialog button which will have an icon if the current system style
+    // settings suggest it:
+    mudlet::self()->mIsIconShownOnDialogButtons = !abort->icon().isNull();
+
+    QTextDocument* pWelcome_document = new QTextDocument(this);
+    if (mudlet::self()->mIsIconShownOnDialogButtons) {
```

Comment:
How about just `mShowIconsOnDialogs`? It would be a less awkward

Reply:
Yeah, I must get over my ***is-Ism***...

Reply:
:innocent: